app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEExtra.kt

```diff
@@ -25,11 +26,12 @@ fun SearchTEActivity.queryDataExtra(savedInstanceState: Bundle?): Map<String, St
                     ?.toList() ?: emptyList()
             if (attributes.size != values.size) return emptyMap()
             attributes.mapIndexed { index, attributeUid ->
-                attributeUid to values[index]
+                attributeUid to listOf(values[index])
             }.toMap()
         }
         savedInstanceState.containsKey(Constants.QUERY_DATA) -> {
-            savedInstanceState.getSerializable(Constants.QUERY_DATA) as Map<String, String>
+            @Suppress("UNCHECKED_CAST")
+            savedInstanceState.getSerializable(Constants.QUERY_DATA) as Map<String, List<String>>
```

Comment:
`getSerializable` is deprecated, can you find another alternative?. Maybe using Parcelable